includes/admin/class-sensei-onboarding-api.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * Onboarding REST API.
+ *
+ * @package Sensei\Onboarding
+ * @since   3.1.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei Onboarding REST API endpoints.
+ *
+ * @package Sensei
+ * @author  Automattic
+ * @since   3.1.0
+ */
+class Sensei_Onboarding_API {
+
+	/**
+	 * Main Onboarding instance.
+	 *
+	 * @var Sensei_Onboarding
+	 */
+	private $onboarding;
+
+	const PURPOSES = [ 'share_knowledge', 'generate_income', 'promote_business', 'provide_certification', 'train_employees', 'other' ];
+
+	/**
+	 * Sensei_Onboarding_API constructor.
+	 *
+	 * @param Sensei_Onboarding $onboarding Onboarding instance.
+	 */
+	public function __construct( $onboarding ) {
+		$this->onboarding = $onboarding;
+	}
+
+	/**
+	 * Register the REST API endpoints for Onboarding.
+	 */
+	public function register() {
+		$common = [
+			'endpoint' => [
+				'permission_callback' => [ $this, 'can_user_access_rest_api' ],
+			],
+			'arg'      => [
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+		];
+
+		$welcome_endpoint = [
+			[
+				'methods'  => 'GET',
+				'callback' => [ $this, 'welcome_get' ],
+			],
+			[
+				'methods'  => 'POST',
+				'callback' => [ $this, 'welcome_submit' ],
+				'args'     => [
+					'usage_tracking' => [
+						'required' => true,
+						'type'     => 'boolean',
+					],
+				],
+			],
+		];
+
+		$purpose_endpoint = [
+			[
+				'methods'  => 'GET',
+				'callback' => [ $this, 'purpose_get' ],
+			],
+			[
+				'methods'  => 'POST',
+				'callback' => [ $this, 'purpose_submit' ],
+				'args'     => [
+					'selected' => [
+						'required' => true,
+						'type'     => 'array',
+						'items'    => [
+							'type' => 'string',
+							'enum' => self::PURPOSES,
+						],
+					],
+					'other'    => [
+						'required' => true,
+						'type'     => 'string',
+					],
+				],
+			],
+		];
+
+		$features_endpoint = [
+			[
+				'methods'  => 'GET',
+				'callback' => [ $this, 'features_get' ],
+			],
+			[
+				'methods'  => 'POST',
+				'callback' => [ $this, 'features_submit' ],
+				'args'     => [
+					'selected' => [
+						'required' => true,
+						'type'     => 'array',
+						'items'    => [
+							'type' => 'string',
+							'enum' => $this->onboarding->plugin_slugs,
+						],
+					],
+				],
+			],
+		];
+
+		Sensei_REST_API_Helper::register_endpoints(
+			'sensei/v1',
```

Comment:
Again, I mentioned this in another PR, but probably best to mention here: what do you think of using a `sensei-internal` namespace instead, to indicate that this is not for external use?

Reply:
Updated in 9bb11657b8ba62091fab7924c2e8191a6522273a